### ISSUE TITLE
Can't boot ReactOS 0.2.5-xbox

### ISSUE THREAD

**Issue by binarymaster** [xqemu/xqemu#248]
# Issue
ReactOS 0.2.5-xbox does not boot to desktop in XQEMU.

# Setup
- A custom [Cromwell 2.41-dev BIOS](https://reactos.org/wiki/Xbox_Port_Tests/Cromwell_2.41-dev#Links) (also [XBE mirror download](https://cdn.discordapp.com/attachments/428399682239332354/598602538148823051/xromwell-2.41-reactos.xbe))
- [ReactOS 0.2.5-xbox file tree](https://svn.reactos.org/downloads/xbox/reactosonxbox.tar.gz) (.tar.gz)
- [ReactOS 0.2.5-xbox image](https://cdn.discordapp.com/attachments/428399682239332354/598627922793725962/xbox_hdd.7z) (.qcow2 ready for XQEMU, legitimately created using **xboxhdm**, shared in the XboxDev Discord)

**This setup is confirmed working (boots to desktop) on real Xbox 1.0, 1.1 and 1.4.**

# Description
In the XQEMU it crashes with unhandled page fault exception in the kernel in `xboxvmp.sys` driver.

I found that it crashes in `XboxVmpMapVideoMemory` when accessing nVidia controller to tell about the framebuffer (this function is called on `IOCTL_VIDEO_MAP_VIDEO_MEMORY`):

https://github.com/reactos/reactos/blob/master/win32ss/drivers/miniport/xboxvmp/xboxvmp.c#L386

## Screenshot

![image](https://user-images.githubusercontent.com/578406/61007425-791dad80-a375-11e9-92f2-754760d24938.png)

## Partially resolved backtrace text
```
Bug detected (code 1e param 0 0 0 0)
  KMODE_EXCEPTION_NOT_HANDLED

Page Fault Exception: 14(2)
Processor: 0 CS:EIP 8:dd0315af <xboxvmp.sys: 15af>
cr2 c0686810 cr3 2950000 Proc: c04b3f28 Pid: 8 <explorer> Thrd: c04c1178 Tid: 2f

DS 10 ES 10 FS 30 GS 10
EAX: 03c00000   EBX: cc9d9840   ECX: cc7ab0f8
EDX: c0686810   EBP: dd997a38   ESI: 0078f538   ESP: dd99795c
EDI: dd997d74   EFLAGS: 00000282 kESP dd99795c kernel stack base dd995000
Eip:
* <xboxvmp.sys:15af (win32ss/drivers/miniport/xboxvmp/xboxvmp.c:387 (XboxVmpMapVideoMemory))>
Frames:
* <xboxvmp.sys:129c (win32ss/drivers/miniport/xboxvmp/xboxvmp.c:172 (XboxVmpStartIO))>
* <videoprt.sys:127e> (IntVideoPortDispatchDeviceControl)
* <ntoskrnl.exe:3376f> (IofCallDriver)
* <win32k.sys:414a> (EngDeviceIoControl)
* <framebuf.DLL:20df> (DrvEnableSurface)
* <win32k.sys:3a133> (IntCreatePrimarySurface)
* <win32k.sys:32c14> (IntInitializeDesktopGraphics)
* <win32k.sys:14cd0> (AddGuiApp)
* <win32k.sys:14dac> (IntGraphicsCheck)
* <win32k.sys:3a315> (IntGdiCreateDC)
* <win32k.sys:28fd8> (DceAllocDCE)
* <win32k.sys:2972d> (NtUserGetDCEx)
* <win32k.sys:28f32> (NtUserGetDC)
* <ntoskrnl.exe:322d> (_interrupt_handler2e)
```

**Comment by binarymaster** [comment]
I also got SuperIO serial debug output from it:
```
Used memory 61440Kb
DriverBase for \SystemRoot\system32\drivers\pci.sys: dceb6000
Peripheral Component Interconnect Bus Driver
DriverBase for scsiport.sys: dcebf000
DriverBase for atapi.sys: dceca000
DriverBase for class2.sys: dced5000
DriverBase for disk.sys: dcede000
DriverBase for vfatfs.sys: dcee6000
DriverBase for bootvid.sys: dcefc000
DriverBase for ndis.sys: dcf1c000
DriverBase for \SystemRoot\system32\drivers\floppy.sys: dcf51000
DriverBase for \SystemRoot\system32\drivers\cdrom.sys: dcf72000
DriverBase for \SystemRoot\system32\drivers\fs_rec.sys: dcf89000
DriverBase for \SystemRoot\system32\drivers\beep.sys: dcf98000
DriverBase for \SystemRoot\system32\drivers\null.sys: dcfae000
DriverBase for \SystemRoot\system32\drivers\serial.sys: dcfbd000
Serial Driver 0.0.2
DriverBase for \SystemRoot\system32\drivers\keyboard.sys: dcfd3000
DriverBase for \SystemRoot\system32\drivers\mouclass.sys: dcfe4000
DriverBase for \SystemRoot\system32\drivers\debugout.sys: dcff2000
DriverBase for \SystemRoot\system32\drivers\blue.sys: dd008000
DriverBase for \SystemRoot\system32\drivers\xboxvmp.sys: dd017000
DriverBase for \SystemRoot\system32\drivers\videoprt.sys: dd035000
WARNING:  VideoPortGetAccessRanges at resource.c:456 is UNIMPLEMENTED!
DriverBase for \SystemRoot\system32\drivers\vbemp.sys: dd050000
DriverBase for \SystemRoot\system32\drivers\vgamp.sys: dd067000
DriverBase for \SystemRoot\system32\drivers\msfs.sys: dd07f000
DriverBase for \SystemRoot\system32\drivers\npfs.sys: dd096000
DriverBase for \SystemRoot\system32\win32k.sys: dd3b6000
DriverBase for \SystemRoot\system32\freetype.dll: dd4b3000
DriverBase for \SystemRoot\System32\kbdus.dll: dd535000
(ntuser/input.c:168) Win32K: Failed to open mouse.
WL-Status: ReactOS is starting up...
(service/sctrl.c:120) ScDispatcherLoop() called
DriverBase for \SystemRoot\system32\drivers\tcpip.sys: dd914000
Here1
kmem_malloc(1)
kmem_malloc done
Freeing c04ae000 onto the free list
Freeing c04ae800 onto the free list
done with m_clalloc
Here2
domaininit starting
(1) Domain internet initializing
Registering protocols for internet
Domaininit done
DriverBase for \SystemRoot\system32\drivers\afd.sys: dd957000
(environment.c:85) GetShortPathNameW() failed (Error 6)
(environment.c:85) GetShortPathNameW() failed (Error 6)
(NTDLL:ldr/utils.c:2060) Relocating (78000000 -> 890000) C:\ReactOS\system32\dnsapi.dll
DriverBase for \SystemRoot\System32\framebuf.DLL: ddaca000
KeBugCheckWithTf at ke/catch.c:173
Bug detected (code 1e param 0 0 0 0)
  KMODE_EXCEPTION_NOT_HANDLED


Page Fault Exception: 14(2)
Processor: 0 CS:EIP 8:dd0185af <xboxvmp.sys: 15af>
cr2 c0686810 cr3 296b000 Proc: c04b2208 Pid: 8 <explorer> Thrd: c04bf448 Tid: 2f
DS 10 ES 10 FS 30 GS 10
EAX: 03c00000   EBX: cc9d9850   ECX: cc7ab4e8
EDX: c0686810   EBP: dd97ea38   ESI: 0078f538   ESP: dd97e95c
EDI: dd97ed74   EFLAGS: 00000282 kESP dd97e95c kernel stack base dd97c000
Frames: <xboxvmp.sys: 129c> <videoprt.sys: 127e> <ntoskrnl.exe: 3376f> <win32k.sys: 414a> <framebuf.DLL: 20df> <win32k.sys: 3a133> <win32k.sys: 32c14> <win32k.sys: 14cd0> <win32k.sys: 14dac> <win32k.sys: 3a315> <win32k.sys: 28fd8> <win32k.sys: 2972d> <win32k.sys: 28f32> <ntoskrnl.exe: 322d> <77E6BC3B> <77E6D8AF> <77E78F21> <77E83CE4> <77E77AA0> <77E78755> <77E779A4> <77E78DD2> <76162DEC> <76169A0E> <7C905360> <7C908C5B> <7C906216> <7C810163> <7C8100FD> <777C15F5> <777C177F> <777C3AB0> <777C4015> <777C4068> <777CB540> <7C905360> <7C908C5B> <7C907E43> <7C9051D4> <7C909947> <DEADBEEF>
```

**Comment by binarymaster** [comment]
Fixed in dd1248a. :slightly_smiling_face:
